src/librustc_codegen_llvm/lib.rs

```diff
@@ -270,13 +270,12 @@ impl CodegenBackend for LlvmCodegenBackend {
         )
     }
 
-    fn join_codegen_and_link(
+    fn join_codegen(
         &self,
         ongoing_codegen: Box<dyn Any>,
         sess: &Session,
         dep_graph: &DepGraph,
-        outputs: &OutputFilenames,
-    ) -> Result<(), ErrorReported> {
+    ) -> Result<Box<dyn Any>, ErrorReported> {
         let (codegen_results, work_products) = ongoing_codegen
             .downcast::<rustc_codegen_ssa::back::write::OngoingCodegen<LlvmCodegenBackend>>()
             .expect("Expected LlvmCodegenBackend's OngoingCodegen, found Box<Any>")
```

Comment:
Could you please move https://github.com/0dvictor/rust/blob/8aff8e1d05620458f748afea6b53fc86ae47c552/src/librustc_codegen_llvm/lib.rs#L306 to rustc_interface?

Reply:
Done.